### DISCUSSION TITLE
Term Clarification

### DISCUSSION THREAD

**Discussion by cswoboda** [GreenleafLab/ArchR#719] [General]
Hello!

I couldn't find anything about these in the ArchR manual so I was hoping someone could provide how a few of these terms are calculated and what their usefulness is as a metric:

ReadsInBlacklist
BlacklistRatio
ReadsInPromoter
PromoterRatio
NucleosomeRatio
nMultiFrags
nMonoFrags
nDiFrags
ReadsinPeaks
FRIP

Thanks! Sorry if this is a novice question, I can assume the utility of some of them but would really like to make sure I'm not making any assumptions about these metrics based off the name.

**Comment by rcorces** [comment]
You can read about blacklists here - https://www.nature.com/articles/s41598-019-45839-z

Explaining the utility of each would take a long time but I'll put it on the to-do list for documentation one day.

ReadsInBlacklist - the number of reads that fall in blacklist regions
BlacklistRatio - the ratio of reads in blacklist regions to reads outside of blacklist regions
ReadsInPromoter - the number of reads that fall in promoter regions
PromoterRatio - the ratio of reads in promoters to reads outside of promoters
NucleosomeRatio - similar to above but for reads mapping to nucleosome-sized reads, defined as:
`(nDiFrags + nMultiFrags) / nMonoFrags)`
ReadsinPeaks - the number of reads falling in peak regions
FRIP - the fraction of reads falling in peak regions

@jgranja24 can correct me if I'm wrong on these:
Given a length of DNA that wraps around a nucleosome (147 bp by default)
nMonoFrags - the number of fragments that have length less than `2 * nucLength`
nDiFrags - the number of fragments that have length less than `3 * nucLength` but greater than `2 * nucLength`
nMultiFrags - the number of fragments that have length >= `3*nucLength`

**Reply by mlin2017** [reply]
Hi cswoboda and rcorces,

How about nucleosome-free fragments?I thought nucleosome-free fragments would be dominant in all fragments from scATAC or ATAC-seq data. If mononucleosome-fragments are dominant in a data rather than nucleosome-free-fragments, what does it mean? Should I worry about the sequencing quality?

Thanks,
Mao

**Reply by rcorces** [reply]
The 10x Genomics standard protocol uses a double-sided bead selection to remove large fragments and leftover primers etc. Depending on the accuracy with which that bead selection is performed, you can lose more of the nucleosome-free/mono nucleosomal fragments. We havent noticed too much of a difference in data quality when this happens.

**Reply by mlin2017** [reply]
Thank you Ryan for your quick reply and welcome back!

It seems that it is safe to say that the fraction of nucleosome-free or monocucleosomal fragment (which one dominates) does not matter much, as long as these two types of fragments are the majority and the distribution of fragment sizes shows a clear periodicity at ~150bp?

Can I ask another question regarding some of these metrics? Here are two scatter plots from our data:
![Screen Shot 2022-01-25 at 1 14 43 PM](https://user-images.githubusercontent.com/25777670/151035133-93b55378-4255-4ff2-a40c-7e762a8e56d0.png)

Left: TSSenrichment score vs (nMonoFrags / (nDiFrags + nMultiFrags) ); right: ReadsInTSS vs nFrags colored by TSSenrichment score. 
It seems that there are optimal ranges of NucleosomeRatio(its reciprocal) and nFrags, highlighted in red circles. We are curious about why we observe those optimal regions, for instance why the yellow patch on the right plot stops there and does not continue along the top-right? Is this kind of general feature of scATAC-seq? Did you notice this similar pattern in other datasets? 

Thanks!

**Reply by rcorces** [reply]
> It seems that it is safe to say that the fraction of nucleosome-free or monocucleosomal fragment (which one dominates) does not matter much, as long as these two types of fragments are the majority and the distribution of fragment sizes shows a clear periodicity at ~150bp?

sounds right to me.

For Q#2 - 

TSS enrichment is a measure of signal-to-background. A dead cell that lacks nucleosomes and just has naked DNA will produce a very high number of unique fragments because its whole genome is accessible to Tn5. But it will have a very low TSS enrichment score because it essentially just has  background transposition. I believe this concept explains your observations - a cell with a good TSS enrichment will have a natural upper bound on how many unique fragments it generates.

**Reply by mlin2017** [reply]
Thanks! The natural upper bound for a good quality cell makes total sense.

I am still not quite sure why cells with high TSSenrichment (the yellow region on the plot on the right) form a bowl-like shape but not a region parallel to and above the linear regression line. Conceptually, TSS enrichment scores can also be calculated by the ratio of #ReadsInTSS/nReads(~nFrags*2). TSS enrichment scores calculated this way are supposed to be highly correlated with the TSSEnrichment metric in CellColData (50bp TSS region vs +/- 1900-2000 flanking region). However, these two ways of calculating TSS enrichment scores do not seem to agree with each other quite well, as shown on the plot. What do you think about the differences between these two methods?

By the way, are those +/- 1900-2000 bp flanking regions used in ArchR calculation  ~100 bp long or ~2000bp long?

Thank you!

**Reply by rcorces** [reply]
> Conceptually, TSS enrichment scores can also be calculated by the ratio of #ReadsInTSS/nReads

No I dont think thats the correct way to think about this. That is equivalent to the fraction of reads in promoters, which we demonstrate in the paper is not as good at discriminating quality and is not equivalent to the TSS score.

> By the way, are those +/- 1900-2000 bp flanking regions used in ArchR calculation ~100 bp long or ~2000bp long?

100 bp.

Maybe this figure helps clarify what a TSS enrichment score is:

![image](https://user-images.githubusercontent.com/10343046/151076057-faf09e07-7bad-41ae-a17f-f14ec03d90b0.png)

**Reply by mlin2017** [reply]
Thank you and this is very helpful!

Could you please point me to the paper(s) where the differences between fraction of reads in promoters and TSS score in discriminating cell quality were demonstrated and where this above figure comes from (if not from the same paper)? I searched a few papers from you and your group but did not these information. Thanks!

**Reply by rcorces** [reply]
> Could you please point me to the paper(s) where the differences between fraction of reads in promoters and TSS score in discriminating cell quality were demonstrated

Supplementary Figure 2 panels c, d, g, h in the ArchR manuscript.

> where this above figure comes from (if not from the same paper)

It is unpublished

**Reply by mlin2017** [reply]
oh, thank you so much!

**Comment by DaianeH** [comment]
Hi @rcorces,
Do you have any suggestions or guideline on thresholds to filter low quality cells using PromoterRatio, BlacklistRatio and NucleosomeRatio?
Thank you in advance!

**Reply by rcorces** [reply]
I do not use those metrics to filter cells. The manual describes the approach I recommend.
